src/functions.js

```diff
@@ -109,16 +175,43 @@ function sales( cars ){
   //   'Ford': 20000,
   //   'Vauxhall': 15000
   // }
-}
+ /*  for(var i=0;i<cars.length;i++){
+    for(var key in cars[i]){ */
+/*       if(cars[i]['make']==='Ford') cars['price'] +=cars[i]['price'];
+      if(cars[i][key]==='Land Rover') cars['price'] +=cars[i]['price'];
+      if(cars[i][key]==='Toyota') cars['price'] +=cars[i]['price'];
+      if(cars[i][key]==='Honda') cars['price'] +=cars[i]['price']; */
+      var newObj={'Ford':0 , 'Land Rover':0 , 'Toyota':0, 'Honda':0};
+      for(var i=0;i<cars.length;i++){
+         for(var key in cars[i]){
+             if(cars[i][key]=='Ford') newObj['Ford'] +=cars[i]['price']
+             //.push(arr[i][key]) //arr[i][key]
+             if(cars[i][key] =='Land Rover') newObj['Land Rover'] +=cars[i]['price']
+             if(cars[i][key]=='Toyota') newObj['Toyota']+=cars[i]['price']
+             if(cars[i][key] =='Honda') newObj['Honda']+=cars[i]['price']
+         }
+      }
+return newObj;
+}//lloc
 
 // Harder challenges
-function secondLargest( numbers ){
+function secondLargest(arr){
   // numbers is an array of numbers
   // return the index of the second 
   // largest number in the array
-}
 
-function factorial( int ) {
+    /* var maxi = Math.max(...arr);
+    arr2= arr.filter(i => i !== maxi);
+    return arr2.indexOf(Math.max(...arr2)); */
+
+   //smart way
+   arrCopy=arr.slice();       //makes a copy of original array
+    var secMax = arrCopy.sort((a, b)=>b - a)[1]
+    return arr.indexOf(secMax); 
+    
+  }
```

Comment:
Very clever solution and way simpler than what I came up with. Mahsa 1 - 0 Stack Overflow...